pyproject.toml

```diff
@@ -70,10 +69,12 @@ build-backend = "hatchling.build"
 managed = true
 dev-dependencies = [
     "pyright==1.1.349",
-    "ruff>=0.1.15",
+    "ruff>=0.2.0",
     "docformatter>=1.7.5",
     "pytest>=8.0.0",
     "coverage>=7.4.1",
+    "typos>=1.18.2",
+    "rich>=13.7.0",
```

Comment:
Perhaps you should just install it as global tool -> https://rye-up.com/guide/tools/

Reply:
I agree, adding ruff as a dependency instead of a dev dependency was a mistake and adding `typos` makes sense since it's part of the CI, but things like `rich` are personal preferences and should not go in the dev deps of the project.

Reply:
Didn't know about global tools, I'll do that.